hazelcast-jet-spring/src/main/java/com/hazelcast/jet/spring/JetSpringServices.java

```diff
@@ -0,0 +1,288 @@
+    ) {
+        return e -> e.filterUsingService(beanServiceFactory(beanName, requiredType), predicate::test);
+    }
+
+    /**
+     * Returns a transform function for the {@link StreamStage#apply(FunctionEx)}
+     * which adds a filter stage using the specified bean.
+     *
+     * @param requiredType the class of the bean
+     * @param predicate    the predicate
+     * @param <T>          the type of the bean
+     */
+    public static <I, T> FunctionEx<StreamStage<I>, StreamStage<I>> filterStreamUsingSpringBean(
+            @Nonnull Class<T> requiredType,
+            @Nonnull BiPredicateEx<? super T, ? super I> predicate
+    ) {
+        return e -> e.filterUsingService(beanServiceFactory(requiredType), predicate::test);
+    }
+
+    /**
+     * Returns a transform function for the {@link StreamStage#apply(FunctionEx)}
+     * which adds a filter stage using the specified bean.
+     *
+     * @param beanName  the name of the bean
+     * @param predicate the predicate
+     * @param <T>       the type of the bean
+     */
+    public static <I, T> FunctionEx<StreamStage<I>, StreamStage<I>> filterStreamUsingSpringBean(
+            @Nonnull String beanName,
+            @Nonnull BiPredicateEx<? super T, ? super I> predicate
+    ) {
+        return e -> e.filterUsingService(beanServiceFactory(beanName), predicate::test);
+    }
+
+    /**
+     * Returns a factory that provides the specified bean as the service object.
+     *
+     * @param beanName     the name of the bean
+     * @param requiredType the class of the bean
+     * @param <T>          the type of the bean
+     */
+    public static <T> ServiceFactory<?, T> beanServiceFactory(@Nonnull String beanName, @Nonnull Class<T> requiredType) {
+        return ServiceFactory.withCreateContextFn(ctx -> new BeanExtractor())
+                             .withCreateServiceFn((c, g) -> g.getBean(beanName, requiredType));
+    }
+
+    /**
+     * Returns a factory that provides the specified bean as the service object.
+     *
+     * @param requiredType the class of the bean
+     * @param <T>          the type of the bean
+     */
+    public static <T> ServiceFactory<?, T> beanServiceFactory(@Nonnull Class<T> requiredType) {
+        return ServiceFactory.withCreateContextFn(ctx -> new BeanExtractor())
+                             .withCreateServiceFn((c, g) -> g.getBean(requiredType));
+    }
+
+    /**
+     * Returns a factory that provides the specified bean as the service object.
+     *
+     * @param beanName the name of the bean
+     * @param <T>      the type of the bean
+     */
+    public static <T> ServiceFactory<?, T> beanServiceFactory(@Nonnull String beanName) {
+        return ServiceFactory.withCreateContextFn(ctx -> new BeanExtractor())
+                             .withCreateServiceFn((c, g) -> g.getBean(beanName));
+    }
+
+    @SpringAware
+    private static final class BeanExtractor {
```

Comment:
It just occurred to me all the `getBean()` methods should check that `context` is not null. 
If it's null then it means `SpringManagedContext` was not injected into Hazelcast `Config` and we should explain how to do it / point to the right documentation. Otherwise it's hard to figure this out on your own. 

Injected of the `SpringManagedContext` works out of the box only when you are using the Spring XML namespace support and "nobody" does that anymore. However it's more complex when you are using Spring Java Config.